src/gui/LocationDialog.cpp

```diff
@@ -981,3 +991,125 @@ void LocationDialog::filterSitesByRegion()
+	{
+		qWarning() << "Already updating...  will not start again until current update is complete.";
+		return;
+	}
+
+	if (progressBar == Q_NULLPTR)
+		progressBar = StelApp::getInstance().addProgressBar();
+	progressBar->setValue(0);
+	progressBar->setRange(0, 0);
+
+	qDebug() << "Updating timezone fixes...";
+
+	QSettings* conf = StelApp::getInstance().getSettings();
+	QUrl URL(conf->value("main/timezone_url", "https://www.stellarium.org/files/timezone.tab").toString());
+	connect(networkManager, SIGNAL(finished(QNetworkReply*)), this, SLOT(downloadComplete(QNetworkReply*)));
+	QNetworkRequest request;
+	request.setUrl(URL);
+	request.setRawHeader("User-Agent", StelUtils::getUserAgentString().toUtf8());
+#if (QT_VERSION<QT_VERSION_CHECK(6,0,0))
+	request.setAttribute(QNetworkRequest::FollowRedirectsAttribute, true);
+#endif
+	downloadReply = networkManager->get(request);
+	connect(downloadReply, SIGNAL(downloadProgress(qint64,qint64)), this, SLOT(updateDownloadProgress(qint64,qint64)));
+
+	updateState = LocationDialog::Updating;
+}
+
+void LocationDialog::downloadComplete(QNetworkReply *reply)
+{
+	if (reply == nullptr)
+		return;
+
+	disconnect(networkManager, SIGNAL(finished(QNetworkReply*)), this, SLOT(downloadComplete(QNetworkReply*)));
+	deleteDownloadProgressBar();
+
+	if (reply->error() || reply->bytesAvailable()==0)
+	{
+		qWarning() << "Error: While trying to access"
+			   << reply->url().toString()
+			   << "the following error occurred:"
+			   << reply->errorString();
+
+		reply->deleteLater();
+		downloadReply = nullptr;
+		updateState = LocationDialog::DownloadError;
+		return;
+	}
+
+	try
+	{
+		QByteArray data = reply->readAll();
+		if (QString(data).endsWith(":EOF")) // Qt5-compatible check
+		{
+			QString tzfFilePath = StelFileMgr::findFile(StelLocationMgr::getTZFFileName());
+			QFile tzfFile(tzfFilePath);
+			if (tzfFile.exists())
+				tzfFile.remove();
+
+			if (tzfFile.open(QIODevice::WriteOnly | QIODevice::Text))
+			{
+				tzfFile.write(data);
+				tzfFile.close();
+			}
+			qWarning() << "Updating timezone fixes are complete... The new data will be apply on next startup.";
+			updateState = LocationDialog::CompleteUpdates;
+		}
+		else
+		{
+			qWarning() << "Downloaded timezone fixes are not completed, restore the data!";
+			updateState = LocationDialog::OtherError;
+		}
+	}
+	catch (std::runtime_error &e)
+	{
+		qDebug() << "Cannot write timezone fixes to file:" << e.what();
+		return;
+	}
+
+	reply->deleteLater();
+	downloadReply = nullptr;
+
+	emit updateTZFComplete();
+}
+
+void LocationDialog::updateDownloadProgress(qint64 bytesReceived, qint64 bytesTotal)
+{
+	if (progressBar == Q_NULLPTR)
+		return;
+
+	int currentValue = 0;
+	int endValue = 0;
+
+	if (bytesTotal > -1 && bytesReceived <= bytesTotal)
+	{
+		//Round to the greatest possible derived unit
+		while (bytesTotal > 1024)
+		{
+			bytesReceived = static_cast<qint64>(std::floor(static_cast<double>(bytesReceived) / 1024.));
```

Comment:
Why so many explicit casts? You already divide a `qint64` by a `double`, the integer would be implicitly promoted to `double`. Then you floor it, which makes the conversion to `double` senseless, since the same result, without any need for rounding and back-conversion, could be obtained from an integer division:
```
bytesReceived /= 1024;
```